eth/api.go

```diff
@@ -0,0 +1,266 @@
+// These methods can be abused by external users and must be considered insecure for use by untrusted users.
+type PrivateMinerApi struct {
+	e *Ethereum
+}
+
+// NewPrivateMinerApi create a new RPC service which controls the miner of this node.
+func NewPrivateMinerApi(e *Ethereum) *PrivateMinerApi {
+	return &PrivateMinerApi{e: e}
+}
+
+// Start the miner with the given number of threads
+func (s *PrivateMinerApi) Start(threads rpc.HexNumber) (bool, error) {
+	s.e.StartAutoDAG()
+	err := s.e.StartMining(threads.Int(), "")
+	if err == nil {
+		return true, nil
+	}
+	return false, err
+}
+
+// Stop the miner
+func (s *PrivateMinerApi) Stop() bool {
+	s.e.StopMining()
+	return true
+}
+
+// SetExtra sets the extra data string that is included when this miner mines a block.
+func (s *PrivateMinerApi) SetExtra(extra string) (bool, error) {
+	if err := s.e.Miner().SetExtra([]byte(extra)); err != nil {
+		return false, err
+	}
+	return true, nil
+}
+
+// SetGasPrice sets the minimum accepted gas price for the miner.
+func (s *PrivateMinerApi) SetGasPrice(gasPrice rpc.Number) bool {
+	s.e.Miner().SetGasPrice(gasPrice.BigInt())
+	return true
+}
+
+// SetEtherbase sets the etherbase of the miner
+func (s *PrivateMinerApi) SetEtherbase(etherbase common.Address) bool {
+	s.e.SetEtherbase(etherbase)
+	return true
+}
+
+// StartAutoDAG starts auto DAG generation. This will prevent the DAG generating on epoch change
+// which will cause the node to stop mining during the generation process.
+func (s *PrivateMinerApi) StartAutoDAG() bool {
+	s.e.StartAutoDAG()
+	return true
+}
+
+// StopAutoDAG stops auto DAG generation
+func (s *PrivateMinerApi) StopAutoDAG() bool {
+	s.e.StopAutoDAG()
+	return true
+}
+
+// MakeDAG creates the new DAG for the given block number
+func (s *PrivateMinerApi) MakeDAG(blockNr rpc.BlockNumber) (bool, error) {
+	if err := ethash.MakeDAG(uint64(blockNr.Int64()), ""); err != nil {
+		return false, err
+	}
+	return true, nil
+}
+
+// PublicTxPoolApi offers and API for the transaction pool. It only operates on data that is non confidential.
+type PublicTxPoolApi struct {
+	e *Ethereum
+}
+
+// NewPublicTxPoolApi creates a new tx pool service that gives information about the transaction pool.
+func NewPublicTxPoolApi(e *Ethereum) *PublicTxPoolApi {
+	return &PublicTxPoolApi{e}
+}
+
+// Status returns the number of pending and queued transaction in the pool.
+func (s *PublicTxPoolApi) Status() map[string]*rpc.HexNumber {
+	pending, queue := s.e.TxPool().Stats()
+	return map[string]*rpc.HexNumber{
+		"pending": rpc.NewHexNumber(pending),
+		"queued":  rpc.NewHexNumber(queue),
+	}
+}
+
+// PublicAccountApi provides an API to access accounts managed by this node.
+// It offers only methods that can retrieve accounts.
+type PublicAccountApi struct {
+	am *accounts.Manager
+}
+
+// NewPublicAccountApi creates a new PublicAccountApi.
+func NewPublicAccountApi(am *accounts.Manager) *PublicAccountApi {
```

Comment:
so for instance, now, it is clear that this should have the accounts selector. 
I knew we were gonna regret that really hasty modularisation of the API.... but hey, hope its not too late. 
Anyhow, before we deprecate it, `personal.newAccount()` as well as `accounts.newAccount()` should be supported. This is easily doable without code duplication, if we just create 2 rpc.Api-s out of accounts.PrivateApi to insert in both namespaces. 
We should already think about how we handle deprecation, likely `Deprecated` should be a field in `rpc.Api`